src/main/scala/treadle/executable/DataStore.scala

```diff
@@ -621,24 +616,62 @@ object DataStore {
   }
 }
 
-trait HasDataArrays {
+trait DataReader {
+  def apply(symbol: Symbol): Big
+  def apply(symbol: Symbol, offset: Int): Big
+}
+
+trait DataWriter {
+  def update(symbol: Symbol, value: Big): Unit
+  def update(symbol: Symbol, offset: Int, value: Big): Unit
+}
+
+trait DataSerializer {
+  def serialize: String
+  def deserialize(jsonString: String): Unit
+}
+
+trait DataPluginHost {
+  def addPlugin(name: String, plugin: DataStorePlugin, enable: Boolean): Unit
+  def removePlugin(name: String): Unit
+  def hasEnabledPlugins: Boolean
+  val plugins : mutable.HashMap[String, DataStorePlugin]   // TODO: change to getter
+  val activePlugins : mutable.ArrayBuffer[DataStorePlugin] // TODO: change to getter
+
+}
+
+trait DataBuffer extends DataReader {
+  def getTime: Long
+}
+
+trait AbstractDataStore extends DataReader with DataWriter with DataSerializer with DataPluginHost {
```

Comment:
Random comment - might be good to have some ScalaDoc here.

Reply:
Do you think it makes sense to split the different functionality expected from our `AbstractDataStore` into different traits, or should all/some of these be unified directly into `AbstractDataStore`?